osbuild/meta.py

```diff
@@ -697,17 +698,23 @@ def list_modules_for_class(self, klass: str) -> List[str]:
         if not module_path:
             raise ValueError(f"Unsupported nodule class: {klass}")
 
-        path = os.path.join(self.path, module_path)
-        modules = filter(lambda f: os.path.isfile(f"{path}/{f}") and not path.endswith(".meta.json"),
-                         os.listdir(path))
+        modules = []
+        for path in self.libdir.dirs:
+            path = os.path.join(path, module_path)
+            modules += [
+                f for f in os.listdir(path)
+                if os.path.isfile(f"{path}/{f}") and not path.endswith(".meta.json")
+            ]
         return list(modules)
 
     def get_module_info(self, klass, name) -> Optional[ModuleInfo]:
         """Obtain `ModuleInfo` for a given stage or assembler"""
 
         if (klass, name) not in self._module_info:
-
-            info = ModuleInfo.load(self.path, klass, name)
+            for path in self.libdir.dirs:
+                info = ModuleInfo.load(path, klass, name)
```

Comment:
`info` would be unset if `self.libdir.dirs` is empty; it can't be empty due to it being built from a `str.split` but maybe we want to assert the invariant in `Libdir.__init__` regardless?